File: test-suite/src/test/java/io/micronaut/docs/server/binding/Point.java
package io.micronaut.docs.server.binding;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.core.annotation.Introspected;
import io.micronaut.core.annotation.ReflectiveAccess;

import java.util.Objects;

@ReflectiveAccess
@Introspected
public class Point {
    private Integer x;
    private Integer y;

    public Point(
        @JsonProperty("x") Integer x, // @JsonProperty for jackson databind
        @JsonProperty("y") Integer y // @JsonProperty for jackson databind
    ) {
        this.x = x;
        this.y = y;
    }

    public Integer getX() {
        return x;
    }

    public void setX(Integer x) {
        this.x = x;
    }

    public Integer getY() {
        return y;
    }

    public void setY(Integer y) {
        this.y = y;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }

        Point point = (Point) o;

        if (!Objects.equals(x, point.x)) {
            return false;
        }
        return Objects.equals(y, point.y);
    }

    @Override
    public int hashCode() {
        int result = x != null ? x.hashCode() : 0;
        result = 31 * result + (y != null ? y.hashCode() : 0);
        return result;
    }
}

File: test-suite-scala/src/test/scala/io/micronaut/docs/server/binding/Point.scala
/*
 * Copyright 2017-2026 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.micronaut.docs.server.binding

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.core.annotation.Introspected
import io.micronaut.core.annotation.ReflectiveAccess

import scala.annotation.meta.field
import scala.annotation.meta.getter
import scala.annotation.meta.param
import scala.beans.BeanProperty

@ReflectiveAccess
@Introspected
case class Point @JsonCreator() (
    @(JsonProperty @param @field @getter)("x") @BeanProperty x: Integer,
    @(JsonProperty @param @field @getter)("y") @BeanProperty y: Integer
)
